test/functional/legacy/038_virtual_replace_spec.lua

```diff
@@ -0,0 +1,56 @@
+-- Test Virtual replace mode.
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('Virtual replace mode', function()
+  setup(clear)
+
+  it('is working', function()
+    -- Make sure that backspace works, no matter what termcap is used.
+    execute('set t_kD=x7f t_kb=x08')
+    feed('ggdGa<cr>')
+    feed('abcdefghi<cr>')
+    feed('jk<tab>lmn<cr>')
+    feed('<Space><Space><Space><Space>opq<tab>rst<cr>')
+    feed('<C-d>uvwxyz<cr>')
+    feed('<esc>gg:set ai<cr>')
```

Comment:
Sometimes a little manual clean-up makes the test easier to read afterwards. You could align the `:set` here with the one in the next line by splitting this line into two:

    feed('<esc>gg')
    execute('set ai')